apps/src/javalab/CommitDialog.jsx

```diff
@@ -225,36 +290,55 @@ function CommitDialogBody({files, notes, onToggleFile, onChangeNotes}) {
       <div style={{...styles.bold, ...styles.filesHeader}}>
         {i18n.saveToBackpack()}
       </div>
-      {files.map(file => (
-        <div key={file.name} style={styles.fileRow}>
-          <label htmlFor={`commit-${file.name}`} style={styles.fileLabel}>
-            {file.name}
-          </label>
-          <input
-            id={`commit-${file.name}`}
-            type="checkbox"
-            checked={file.commit}
-            onChange={() => onToggleFile(file.name)}
-            style={styles.checkbox}
+      {files.map(file => {
+        return (
+          <CommitDialogFile
+            file={file}
+            onToggleFile={onToggleFile}
+            key={file.name}
           />
-        </div>
-      ))}
+        );
+      })}
     </div>
   );
 }
 
 CommitDialogBody.propTypes = {
-  files: PropTypes.arrayOf(
-    PropTypes.shape({
-      name: PropTypes.string.isRequired,
-      commit: PropTypes.bool.isRequired
-    })
-  ).isRequired,
+  files: PropTypes.arrayOf(fileShape).isRequired,
   notes: PropTypes.string,
   onToggleFile: PropTypes.func.isRequired,
   onChangeNotes: PropTypes.func.isRequired
 };
 
+export function CommitDialogFile({file, onToggleFile}) {
```

Comment:
I'm not super up on the react patterns, should this be its own file? Mainly because this file is getting long and it seems like a clean thing to factor out.

Reply:
Yeah probably should, we have another sub-component already in this file so I felt like I should either factor both out or neither.

Reply:
makes sense, feel free to do as a separate pr if that's easier

Reply:
Think I'll do this as a follow-up, item here:

https://codedotorg.atlassian.net/browse/CSA-791

Reply:
sounds good!